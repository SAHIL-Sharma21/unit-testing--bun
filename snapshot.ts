/**
 * Here i am going to learn the snapshot test
 * It usually do it in frontend .
 */

export function fetchData(userId?: number){
    return {
        id: userId,
        name: "john Doe",
        roles: ['user', 'admin'],
        lastLogin: new Date().toISOString(), // here will give error. give exact date
        preferences: {
            notifications: true,
            theme: "Dark",
        }
    };
}
