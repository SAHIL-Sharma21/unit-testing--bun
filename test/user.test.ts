// snapshot testing
import {describe, test, it, expect} from 'bun:test'
import assert from 'node:assert'
import { fetchData } from '../snapshot';

describe("user feature", () => {
    it("fetched the data from the server", () => {
        const data = fetchData(1);
        expect(data).toMatchSnapshot();
    })
});