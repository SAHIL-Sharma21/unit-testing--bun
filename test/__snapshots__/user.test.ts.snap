// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`user feature fetched the data from the server 1`] = `
{
  "id": 1,
  "lastLogin": "2025-06-08T16:37:22.567Z",
  "name": "john Doe",
  "preferences": {
    "notifications": true,
    "theme": "Dark",
  },
  "roles": [
    "user",
    "admin",
  ],
}
`;
